reana_db/models.py

```diff
@@ -562,24 +598,20 @@ def workflow_status_change_listener(workflow, new_status, old_status, initiator)
     if workflow.run_started_at and finished_at:
         cpu_time = finished_at - workflow.run_started_at
         cpu_milliseconds = int(cpu_time.total_seconds() * 1000)
-        cpu_resource = Resource.query.filter_by(
-            name=DEFAULT_QUOTA_RESOURCES["cpu"]
-        ).one_or_none()
-        if cpu_resource:
-            from .database import Session
-
-            workflow_resource = WorkflowResource(
-                workflow_id=workflow.id_,
-                resource_id=cpu_resource.id_,
-                quantity_used=cpu_milliseconds,
-            )
-            user_resource_quota = UserResource.query.filter_by(
-                user_id=workflow.owner_id, resource_id=cpu_resource.id_
-            ).first()
-            user_resource_quota.quota_used += cpu_milliseconds
-            Session.add(workflow_resource)
-            Session.commit()
+        cpu_resource = get_default_quota_resource(ResourceType.cpu.name)
+        workflow_resource = WorkflowResource(
+            workflow_id=workflow.id_,
+            resource_id=cpu_resource.id_,
+            quantity_used=cpu_milliseconds,
+        )
+        user_resource_quota = UserResource.query.filter_by(
+            user_id=workflow.owner_id, resource_id=cpu_resource.id_
+        ).first()
+        user_resource_quota.quota_used += cpu_milliseconds
+        Session.add(workflow_resource)
+        Session.commit()
         update_users_disk_quota(user=workflow.owner)
+        calculate_disk_workflow_resource(workflow)
```

Comment:
What do you think of changing this function name to be more in harmony with the previous?
```suggestion
        store_workflow_disk_quota(workflow)
```
(or similar)